### PR TITLE
replace sleep with a wait_for loop

### PR DESCRIPTION
**Pull request by jdsn**
drop another static wait and two curl requests with 60s timeout in favor of a wait_for loop to wait until the crowbar api is available

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -1662,14 +1662,7 @@ EOF
         service smb restart
     fi
 
-    sleep 20
-    if ! curl -m 59 -s $crowbar_api  > /dev/null || \
-        ! curl -m 59 -s --digest --user crowbar:crowbar $crowbar_api  | \
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: this drops the part that uses digest auth so we might only learn about problems there later

Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: as the function onadmin_is_crowbar_api_available is already in use (and works) this rather means that this function is missing the digest case (and it was not needed for the current usages of this function), right?

**Comment by bmwiedemann** [comment]
+1

**Comment by MaximilianMeister** [comment]
+1

**Pull request merged**